images/kor/tests/main.tf

```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_exec_test" "version" {
+  digest = var.digest
+  script = "docker run --rm $IMAGE_NAME version"
```

Comment:
Does upstream have a helm chart that can deploy this?

Reply:
The upstream doesn't publish the chart in a helm repo but has it in Git, I'm checking how can we use that.

Reply:
Looks like the upstream maintainer hasn't yet published the helm chart, but it is located in Git, so we could consume it from here:
- https://github.com/yonahd/kor/tree/main/charts/kor

However i'm not sure the ["helm_release"](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release) terraform provider supports referencing charts stored in Git, from what I can see.

I'd suggest having a look to see if there are other similar use-cases where we've already done this, and maybe drop a message in the `chainguard-images` space. it's quite likely we've already ran into this.

Failing that, we could possibly use "null_resource" to run a git clone, then use helm_release to reference the local path of the chart perhaps?

Reply:
You could checkout the repo as part of a script then install it with regular `helm install`. We have some image tests that do `helm install` from a script.

The Helm provider does support local charts, however: https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#example-usage---local-chart